apps/main/lib/main/repositories/articles_repo.rb

```diff
@@ -0,0 +1,14 @@
+require "blog/repository"
+require "main/entities/article"
+
+module Main
+  module Repositories
+    class ArticlesRepo < Blog::Repository[:articles]
+      struct_namespace Main::Entities
+
+      def home_page_listing
```

Comment:
I wonder if there is a better naming opportunity here? I would imagine that "home_page" doesn't have a direct correlation to `ArticlesRepo` and perhaps it shouldn't know anything about a home page.